### PR TITLE
Enable searching tag keys in query builder

### LINKED ISSUE #2296: add search in tag key drop down
as a data explorer
with viewer permissions or higher
when i click the tag names drop down in a filter box
then i expect the ability to search through the tag names

### PR DESCRIPTION
**Pull request by chnn**
Closes #2296 

keys

### CODE REVIEW

File: ui/src/shared/components/SearchableDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            DropdownMenuColors.Onyx
+          }`}
+          style={this.menuStyle}
+        >
+          <FancyScrollbar
+            autoHide={false}
+            autoHeight={true}
+            maxHeight={maxMenuHeight}
+          >
+            <div className="dropdown--menu">
```
Comment (OfTheDelmer) [inline ui/src/shared/components/SearchableDropdown.tsx]: Could this have been a clockface dropdown using an `DropdownMode.ActionList`?

File: ui/src/shared/components/SearchableDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+  }
+
+  public static Button = DropdownButton
+  public static Item = DropdownItem
+  public static Divider = DropdownDivider
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
```
Comment (OfTheDelmer) [inline ui/src/shared/components/SearchableDropdown.tsx]: How do you feel about removing the constructor in favor of  `public state: State = { expanded: false}`

File: ui/src/shared/components/SearchableDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  searchPlaceholder?: string
+  onSearchChange?: (value: string) => void
+}
+
+interface State {
+  expanded: boolean
+}
+
+@ErrorHandling
+export default class SearchableDropdown extends Component<Props, State> {
```
Comment (OfTheDelmer) [inline ui/src/shared/components/SearchableDropdown.tsx]: Could be a good clockface component

File: ui/src/shared/reducers/v2/timeMachines.ts [hunk 4]
```diff
@@ -513,11 +514,19 @@ export const timeMachineReducer = (
...
-        draftState.queryBuilder.tags[index].searchTerm = searchTerm
+        draftState.queryBuilder.tags[index].valuesSearchTerm = searchTerm
+      })
+    }
+
+    case 'SET_BUILDER_KEYS_SEARCH_TERM': {
+      return produce(state, draftState => {
+        const {index, searchTerm} = action.payload
+
+        draftState.queryBuilder.tags[index].keysSearchTerm = searchTerm
```
Comment (OfTheDelmer) [inline ui/src/shared/reducers/v2/timeMachines.ts]: I like these new draftState updates

**Comment by chnn** [comment]
@OfTheDelmer I refactored a bit with a slightly different approach to the `Dropdown` duplication, lemme know what you think.

**Pull request merged**